src/dmd/argtypes.d

```diff
@@ -41,6 +42,28 @@ TypeTuple toArgTypes(Type t)
     public:
         TypeTuple result;
 
+        /*****
+         * Pass type in memory (i.e. on the stack), a tuple of one type, or a tuple of 2 types
+         */
+        void memory()
+        {
+            //printf("\ttoArgTypes() %s => [ ]\n", t.toChars());
+            result = new TypeTuple(); // pass on the stack
+        }
+
+        ///
+        void oneType(Type t)
+        {
+            result = new TypeTuple(t);
+        }
+
+        ///
+        void twoTypes(Type t1, Type t2)
```

Comment:
Am I having deja vu? I thought this was already added as part of another refactoring pr. Must have been somewhere else.

Reply:
I moved them up a level so they would be accessible from other functions.